### DISCUSSION TITLE
selector resolved to hidden error

### DISCUSSION THREAD

**Discussion by InduKrish** [testing-library/playwright-testing-library#530] [Q&A]
while trying to click the checkbox using the following test, it fails with the following error:  selector resolved to hidden <title>check</title>

 async checkActiveStatusdom(text) {
        const header = await this.screen.findByTestId('erow-GroupCode-0');
        console.log(" header" + header)
        const base = this.within(header).getByText("LINEHOLDER")
        console.log("base value" + base);
        const check = this.within(header).getByText("check")
        console.log(" value " + check)
        await expect(base).toContainText(text);
        console.log("text assertion successful")
        await check.click();  ---> fails

}

output:

 headerLocator@query-by-test-id=["erow-GroupCode-0"]
base valueLocator@query-by-test-id=["erow-GroupCode-0"] >> get-by-text=["LINEHOLDER"]
 value Locator@query-by-test-id=["erow-GroupCode-0"] >> get-by-text=["check"]
text assertion successful
  1) [chromium] › tests/web/vacation.bidaward.defaults.spec.po.new.js:14:9 › Bid and award Default page verification › Default page verification 

    Test timeout of 30000ms exceeded.

    locator.click: Target closed
    =========================== logs ===========================
    waiting for selector "query-by-test-id=["erow-GroupCode-0"] >> get-by-text=["check"]"
      selector resolved to hidden <title>check</title>
    attempting click action
      waiting for element to be visible, enabled and stable
        element is not visible - waiting...
    ============================================================

<img width="1773" alt="Screen Shot 2022-09-20 at 11 23 05 AM" src="https://user-images.githubusercontent.com/113629123/191312720-50f1a540-e915-4138-96ae-57e23ff49b6d.png">

<img width="1195" alt="Screen Shot 2022-09-20 at 11 25 30 AM" src="https://user-images.githubusercontent.com/113629123/191312843-ef935ef6-2edc-4829-980e-6442394817be.png">

**Comment by jrolfs** [comment]
It looks like you're attempting to click on the SVG `<title />` element, which is not a visible element. I suggest you review the following documentation to get a better handle on the general Testing Library philosophy and how it is intended to be used to help you test your application more closely to how your users interact with it:

- https://testing-library.com/docs/guiding-principles
- https://testing-library.com/docs/dom-testing-library/faq
- https://testing-library.com/docs/dom-testing-library/api-accessibility

In this case, ideally, you'd instead be querying for the native checkbox control using an accessible label, like so:

```ts
const header = await this.screen.findByTestId('erow-GroupCode-0');
const check = this.within(header).getByRole("checkbox");

await check.click();
```

Once you're reviewed that, if you're still having this kind of problem, it's really an issue with how you're using Playwright, and it has nothing to do with Playwright Testing Library. I suggest you try to debug your tests using Playwright's robust debugging capabilities before opening an issue here:

- https://playwright.dev/docs/debug#playwright-inspector
- https://playwright.dev/docs/debug#actionability-logs

**Comment by InduKrish** [comment]
> It looks like you're attempting to click on the SVG `<title />` element, which is not a visible element. I suggest you review the following documentation to get a better handle on the general Testing Library philosophy and how it is intended to be used to help you test your application more closely to how your users interact with it:
> 

 SVG `<title />` element - checkbox is visible on the webpage, and it works fine with regular playwright code, can you please clarify what you mean by saying that it is not a visible element while using 

const check = this.within(header).getByText("check")
await check.click(); ---> fails

meanwhile i will try this and keep you posted using getByRole()

const header = await this.screen.findByTestId('erow-GroupCode-0');
const check = this.within(header).getByRole("checkbox");

await check.click();

**Comment by jrolfs** [comment]
The `<svg />` element is visible, but `<title />` is an inherently invisible element (visible only to screen readers:

✅ Query + click SVG _using_ `<title />` as accessible name:

> If you _really_ want to click the `<svg />`, this is probably what you want, but since it's a stylized checkbox I think you really do want to query the checkbox itself as I recommended

https://github.com/testing-library/playwright-testing-library/blob/d5a665ba6d62c26e2dcdf83cca86a2a13ce219a6/test/fixture/locators.test.ts#L122-L128

❌ Query + click `<title />` _within_ `<svg />`:

https://github.com/testing-library/playwright-testing-library/blob/d5a665ba6d62c26e2dcdf83cca86a2a13ce219a6/test/fixture/locators.test.ts#L130-L139

-------

#### Markup

https://github.com/testing-library/playwright-testing-library/blob/d5a665ba6d62c26e2dcdf83cca86a2a13ce219a6/test/fixtures/page.html#L30-L38
